### PR TITLE
[BE][API] User Create Delete Update

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
## Issue Link
[LMS-355 [BE][API] User Create, Delete, Update](https://framgiaph.backlog.com/view/LMS-355)

## Defintion of Done
* [x] create an api to add or create a user
* [x] store user under a company on create depending on its role (trainee or trainer)
* [x] create an api to delete a user (soft delete, doesnt delete the record itself, but chnage is_active to false resulting to invalid token on use)
* [x] create an api to update user details
* [x] must have BE validations
* [x] move every logic in serializer

## Steps to reproduce
* first clear your DB and do migrations and seeder
* login using the admin creds to get token
* use the token for create api

### Create
* pass required fields (see the screenshots below)
* response must be the user data

### Update
* pass required fields (see the screenshots below)
* response must be the updated data

### Delete
* pass the user ID on second param of the URL (/api/users/<:COMPANY_ID>/<:USER_ID>)
* response is just a message
* to test if delete is succesful, login the deleted user, and use its token to any API request, the token should be invalid.

## Affected Components / Functionalities / Page
* serializer
* urls
* views
* migration
* model

## Test Cases
_will update soon..._

## Notes
n/a

### Create
image

### Update
image

### Delete
image

### Test Delete
image
image

### CODE REVIEW

**Review by mikorosales — APPROVED** [review]
APPROVED
- [x] POST - Working
- [x] PUT - Working
- [x] DELETE - Working

Comments: 0
Test time: 40mins
Iterations: 0

**Review by FranzDelosSantos — COMMENTED** [review]
Ster please update the PR with input field validation  according to the data dictionary.
https://docs.google.com/spreadsheets/d/1M0Pf5ssVEgc4vh-6CpCPgcl-1guAz4J9WGJG2o5Xr8M/edit#gid=0
The user will still be created and updated even though some fields are invalid
prError1
prError2

**Comment by ragg09-sunasterisk** [comment]
Validation Fixes
image

**Review by FranzDelosSantos — APPROVED** [review]
## Approved

- [x] create an api to add or create a user
- [x] create an api to delete a user (soft delete)
- [x] create an api to update user details
- [x] must have BE validations

Comments: 1
Test time: 40 mins
Iterations: 1

### Success
prGood3
prGood4

### Vallidation
prGGood

### Soft Delete
prGood2

File: api/app_sph_lms/utils/enum.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+class UserRoleEnum(Enum):
+    ADMIN = 1
+    COMPANY = 2
+    TRAINER = 3
+    TRAINEE = 4
+    
+class StatusEnum(Enum):
+    ACTIVE = 1
+    INACTIVE = 2
```
Comment (mikorosales) [inline api/app_sph_lms/utils/enum.py:11]: no new line

**Pull request merged**